### PR TITLE
Add circleci config build

### PR DESCRIPTION
**Pull request by spirosoik**

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+machine:
+  pre:
+    - cd /opt/circleci/.pyenv/plugins/python-build/../.. && git pull && cd -
```
Comment (owais) [inline circle.yml:3]: Do we need this? CI automatically checks out code

File: circle.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  python:
+    version: 3.5.5
+  timezone:
+    America/Los_Angeles
+
+dependencies:
+  override:
+    - pip install -U pip wheel
+    - make
+    - pyenv local 2.7.11 3.5.5
```
Comment (owais) [inline circle.yml]: Do we want to support 2.7? I suggest we only support Py3

**Review by owais — APPROVED** [review]

File: circle.yml (hunk 1 above)
Comment (spirosoik) [reply circle.yml:3]: CI asks for this :)

File: circle.yml (hunk 2 above)
Comment (spirosoik) [reply circle.yml]: We can remove. Olekisy added back in `tox.ini` and I fixed the broken CI

File: circle.yml (hunk 1 above)
Comment (owais) [reply circle.yml:3]: It didn't before and in other projects :)  but anyway, if it works then no need to change.

File: circle.yml (hunk 2 above)
Comment (owais) [reply circle.yml]: I'm suggesting we don't support 2.7 at all.

**Review by owais — APPROVED** [review]

File: circle.yml (hunk 1 above)
Comment (spirosoik) [reply circle.yml:3]: For some reason circleci wasn't able to find `3.5.5`

File: circle.yml (hunk 2 above)
Comment (owais) [reply circle.yml]: On second thought, it already works and we don't gain much from removing it. May be we should keep it.

**Pull request merged**